src/codeunit/SpotifyService/ITICreateSpotifyPlaylist.Codeunit.al

```diff
@@ -0,0 +1,178 @@
+        RequestURI := 'https://api.spotify.com/v1/users/31rkjqx2c7vxjbyjhtyefxs4232q/playlists';
+
+        CreateJsonString(JsonString, PlaylistName, PlaylistDesc);
+        ITIHttp.PrepareAndSendRequest(HttpRequestMessage, HttpClient, HttpResponseMessage, RequestMethod, RequestURI, JsonString);
+
+        if not HttpResponseMessage.IsSuccessStatusCode() then
+            Message(HttpResponseMessage.ReasonPhrase())
+
+        else begin
+            HttpResponseMessage.Content.ReadAs(ResponseJsonString);
+            JsonObjResponse.ReadFrom(ResponseJsonString);
+            JsonObjResponse.Get('id', PlaylistIDJasonToken);
+            PlaylistID := PlaylistIDJasonToken.AsValue().AsText();
+        end;
+    end;
+
+    local procedure GetArtistTopTracksUris(var ArtistsIDList: List of [Text]; var TracksIDString: Text)
+    var
+        RequestMethod: Text;
+        ArtistID: Text;
+    begin
+        RequestMethod := 'GET';
+        foreach ArtistID in ArtistsIDList do
+            GetTopTracks(TracksIDString, ArtistID, RequestMethod);
+    end;
+
+    local procedure GetArtist(var ArtistsID: List of [Text]; var ArtistJsonObjList: List of [JsonObject]; Artist: Text)
+    var
+        ITIHttp: Codeunit "ITI Http";
+        HttpRequestMessage: HttpRequestMessage;
+        HttpResponseMessage: HttpResponseMessage;
+        HttpClient: HttpClient;
+        RequestContentString: Text;
+        RequestMethod: Text;
+        RequestURI: Text;
+        JsonObj: JsonObject;
+        ArtistIDJasonToken: JsonToken;
+        ArtistNameJasonToken: JsonToken;
+        ArtistIDText: Text;
+        ArtistNameText: Text;
+        JsonString: Text;
+        queryid: Text;
+        queryname: Text;
+        ArtistJsonObj: JsonObject;
+    begin
+        RequestMethod := 'GET';
+        RequestURI := 'https://api.spotify.com/v1/search?q=' + Artist + '&type=artist&limit=1';
+        ITIHttp.PrepareAndSendRequest(HttpRequestMessage, HttpClient, HttpResponseMessage, RequestMethod, RequestURI, RequestContentString);
+        HttpResponseMessage.Content.ReadAs(JsonString);
+        JsonObj.ReadFrom(JsonString);
+        queryid := '$.artists.items[0].id';
+        queryname := '$.artists.items[0].name';
+        if not JsonObj.SelectToken(queryid, ArtistIDJasonToken) then
+            exit;
+        JsonObj.SelectToken(queryname, ArtistNameJasonToken);
+        ArtistIDText := ArtistIDJasonToken.AsValue().AsText();
+        ArtistNameText := ArtistNameJasonToken.AsValue().AsText();
+        ArtistJsonObj.Add('id', ArtistIDText);
+        ArtistJsonObj.Add('name', ArtistNameText);
+        ArtistJsonObjList.Add(ArtistJsonObj);
+        ArtistsID.Add(ArtistIDText);
+    end;
+
+    procedure IterateArtists(var Artists: List of [Text]; var ArtistsIDList: List of [Text]; var ArtistJsonObjList: List of [JsonObject])
+    var
+        Artist: Text;
+    begin
+        foreach Artist in Artists do
+            GetArtist(ArtistsIDList, ArtistJsonObjList, Artist);
+    end;
```

Comment:
W nawiązaniu do poprzednich komentarzy, funkcja `IterateArtists` przyjmowała by parametr `var ITIArtist: Record "ITI Artist"`, a implementacja mogłaby wyglądać tak:

`
begin
    if ITIArtist.FindSet() then
        repeat
             GetArtist(ArtistsIDList, ArtistJsonObjList, ITIArtist.Name);
        until ITIArtist.Next() = 0;
end;
`

Reply:
Proponuję też zmienić nazwę tej funkcji na ```GetArtists``` lub ```GetArtistsDetails``` oraz przenieść tę funkcję nad funkcję ```GetArtist``` (której nazwa, w przypadku zmiany nazwy tej funkcji na `GetArtistsDetails`, powinna być `GetArtistDetails`)

Dlaczego tak? Jak piszemy kod na projekcie, to najczęściej ustawiamy funkcje w kolejności takiej, w jakiej są wywoływane, jest dzięki temu trochę czytelniej i wygodniej, bo czytająć funkcję X która wywołuje funkcje Y, wiesz, że funkcja Y będzie pod spodem

Reply:
I kolejna uwaga - może zamiast robić listę JsonObject-ów, to zrobić JsonArray? Będzie trochę bardziej naturalnie, a w JsonArray wciąż będziesz mógł zrobić
```
foreach JObject in JArray do
    // ...
```

Reply:
I jeszcze jedna rzecz, wydaje mi się, że byłoby trochę lepiej, gdyby funkcja `GetArtist` tylko uzupełniała wartości w wysłanych referencjach (parametrach), ale samo dodawanie do listy czy innej struktury, robił bym już w GetArtists() (która się obecnie nazywa IterateArtists)

Bo gdy ktoś widzi funkcję o nazwie GetArtist, to myśli że dostanie tylko jednego artystę, to zacznie się zastanawiać, po co musi wysyłać aż listę a nie tylko pojedynczą zmienną